### PR TITLE
Fix native library loading

### LINKED ISSUE #128: Remove NativeLibraryLoader paths on DNX
~~Blocked on aspnet/dnx#402~~
~~Blocked on aspnet/dnx#3119~~

### LINKED ISSUE #134: Unable to load DLL 'sqlite3' when using packages.config
Because NuGet only copies runtime artifacts when using `project.json`, if you try to use the package on a project using `packages.config`, you'll get the following error.

> Unable to load DLL 'sqlite3': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

We might be able to make this work using MSBuild scripts...

### PR DESCRIPTION
**Pull request by bricelam**
Also includes cleanup from the dnx451 to net451 change and other minor refactorings.

Resolves #128, fixes #134

### CODE REVIEW

File: src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs [hunk 1]
```diff
@@ -4,40 +4,19 @@
...
 
 namespace Microsoft.Data.Sqlite.Interop
 {
     internal static class NativeMethods
     {
-#if NET451 || DOTNET5_4
+#if NET451
         static NativeMethods()
         {
             var loaded = NativeLibraryLoader.TryLoad("sqlite3");
```
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: For which scenarios do we still need this?

File: src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs [hunk 2]
```diff
@@ -103,6 +80,8 @@ public static bool IsLoaded(string dllName)
             return handle != IntPtr.Zero;
         }
 
+        private static bool IsDNX() => PlatformServices.Default != null;
```
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: Won't this always be true with its [current implementation](https://github.com/aspnet/dnx/blob/0b4140fd6827acc912eed858431922574cd3f9f2/src/Microsoft.Extensions.PlatformAbstractions/PlatformServices.cs#L9-L17)?

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: Hmm, it seems to work, but I'll change it to `PlatformServices.Default?.Application.RuntimeFramework.Identifier == "DNX"` just to be safe.

File: src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs (hunk 1 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: Full .NET (non-DNX)

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: AnyCPU doesn't know its architecture until runtime.

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: Ok, that’s what I was missing. Can we add into the comments?The MSBuild fixes this for full .NET when platform arch is known, right?

File: src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs (hunk 2 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: Ah, it changed in RC2... That's why it's currently working for me :smile:

File: src/Microsoft.Data.Sqlite/SqliteDataReader.cs [hunk 3]
```diff
@@ -128,11 +125,6 @@ public override bool NextResult()
 #if NET451
         // TODO: Remove when fixed in System.Data.Common
         public override void Close() => Dispose(true);
-
-        public override DataTable GetSchemaTable()
```
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteDataReader.cs]: Unfortunately we still need all these NET451 implementations b/c we build against Mono for testing.

File: src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs (hunk 1 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs]: It currently *always* copies both the x86 and x64 version and loads the correct one at runtime. We could optimize it to copy the architecture-specific one if it's know at build time, but I didn't bother...

File: src/Microsoft.Data.Sqlite/SqliteDataReader.cs (hunk 3 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteDataReader.cs]: Ah, is that what the failures are? I thought they updated to net451 contracts...

**Comment by natemcmaster** [comment]
:shipit: (assuming you have approval)

**Pull request merged**